src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/classmemberimpliedmodifier/InputClassMemberImpliedModifierRecords.java

```diff
@@ -0,0 +1,34 @@
+//non-compiled with javac: Compilable with Java14
+package com.puppycrawl.tools.checkstyle.checks.modifier.classmemberimpliedmodifier;
+/* Config:
+ * violateImpliedStaticOnNestedEnum = true
+ * violateImpliedStaticOnNestedInterface = true
+ * violateImpliedStaticOnNestedRecord = true
+ */
+public class InputClassMemberImpliedModifierRecords {
+    public static interface GoodInterface {} // OK
+    // Implied modifier 'static' should be explicit. [ClassMemberImpliedModifier]
+    public interface BadInterface {}
+
+    public static enum GoodEnum {} // OK
+    // Implied modifier 'static' should be explicit. [ClassMemberImpliedModifier]
+    public enum BadEnum {}
+
+    public static record GoodRecord() {} // OK
+    // Implied modifier 'static' should be explicit. [ClassMemberImpliedModifier]
+    public record BadRecord() {}
+
+    public static record OuterRecord() {
```

Comment:
@romani we are checking for implied modifiers _inside_ of records, as well; this is why I changed the documentation to say "classes and records".